prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -85,6 +91,73 @@ class ReportFunction(
         }
     }
 
+    /**
+     * GET report to download
+     *
+     * @see ../../../docs/api/reports.yml
+     */
+    @FunctionName("getMessagesFromTestBank")
+    fun downloadReport(
+        @HttpTrigger(
+            name = "downloadReport",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
```

Comment:
I think it would be better to use an azure function key to call this API, since we will have to manually call this API every time we need a report and we won't be using the front end to call this API. It will make it easier to use and we wouldn't have to get a bearer token every time we wanted to use it.
There's an example inside `ReportFunction.kt` in line 70. You'd just have to set to authLevel to `authLevel = AuthorizationLevel.FUNCTION` and once it's in staging and prod we can generate a random key for it.
If we do it that way we can remove the authentication logic and just rely on azure's authorization.